### PR TITLE
feat(permissions-provider): initial permissions provider

### PR DESCRIPTION
**Pull request by zouxuoz**

### CODE REVIEW

File: packages/permissions-provider/src/getPermissions.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+export const getPermissions = R.pipe(
+  R.pathOr([], ['user', 'roles', 'items']),
+  R.map(
+    R.pipe(
+      R.pathOr([], ['permissions', 'items']),
+      R.groupBy(R.prop('resourceType')),
+      R.mapObjIndexed(R.indexBy(R.prop('resource'))),
+    ),
+  ),
+  R.reduce((result, value) => R.mergeDeepWithKey((k, l, r) => r || l, result, value), {}),
```
Comment (ej9x) [inline packages/permissions-provider/src/getPermissions.js]: k, l, r? Maybe spell out it?

File: packages/permissions-provider/test/__tests__/__snapshots__/index.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`As a developer, I can use \`IfAllowed\` component for conditional rendering based user permissions. 1`] = `"Allowed"`;
+
+exports[`As a developer, I can use \`IfAllowed\` component for pass permission check result via render props. 1`] = `"Allowed = true"`;
+
+exports[`As a developer, I can use \`isAllowed\` for check access via context. 1`] = `"Allowed = false"`;
```
Comment (ej9x) [inline packages/permissions-provider/test/__tests__/__snapshots__/index.test.js.snap]: Maybe use there an inline snapshot? What do you think?

Comment (zouxuoz) [reply packages/permissions-provider/test/__tests__/__snapshots__/index.test.js.snap]: good idea, fixed

File: packages/permissions-provider/src/getPermissions.js (hunk 1 above)
Comment (zouxuoz) [reply packages/permissions-provider/src/getPermissions.js]: fixed

**Pull request merged**